rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -2768,11 +2745,11 @@ private boolean verifyLockDoesNotSurpassLockingCap(BtcTransaction btcTx, Coin to
         }
 
         Coin fedCurrentFunds = getBtcLockedInFederation();
-        Coin lockingCap = this.getLockingCap();
+        Optional<Coin> lockingCap = lockingCapSupport.getLockingCap();
         logger.trace("Evaluating locking cap for: TxId {}. Value to lock {}. Current funds {}. Current locking cap {}", btcTx.getHash(true), totalAmount, fedCurrentFunds, lockingCap);
         Coin fedUTXOsAfterThisLock = fedCurrentFunds.add(totalAmount);
         // If the federation funds (including this new UTXO) are smaller than or equals to the current locking cap, we are fine.
-        if (fedUTXOsAfterThisLock.compareTo(lockingCap) <= 0) {
+        if (lockingCap.isPresent() && fedUTXOsAfterThisLock.compareTo(lockingCap.get()) <= 0) {
```

Comment:
This is a bit confusing to me, it implies that lockingCap value could not be present and in that case it will return `false` indicating that the locking cap exceeded.

Reply:
Done!